kafka_utils/kafka_check/commands/under_replicated.py

```diff
@@ -0,0 +1,135 @@
+                 'not first brokers from Kafka cluster. Set --broker-id to -1 to read broker-id '
+                 'from --data-path. Default: %(default)s',
+        )
+        subparser.add_argument(
+            '--jolokia-port',
+            type=convert_to_port,
+            default=DEFAULT_JOLOKIA_PORT,
+            help='Port to access to jolokia on each broker. Default: %(default)s',
+        )
+
+        return subparser
+
+    def run_command(self):
+        """Under_replicated command, checks number of under replicated partitions for
+        all brokers in the Kafka cluster."""
+        broker_list = self.zk.get_brokers()
+
+        if self.args.first_broker_only:
+            if self.args.broker_id is None:
+                return status_code.WARNING, 'Broker id is not specified'
+
+            if not check_run_on_first_broker(broker_list, self.args.broker_id, self.args.data_path):
+                return status_code.OK, 'Provided broker is not the first in broker-list'
+
+        count = get_under_replicated(broker_list, self.args.jolokia_port)
+
+        if count == 0:
+            return status_code.OK, 'No under replicated partitions'
+        else:
+            msg = "{under_replicated} under replicated partitions".format(under_replicated=count)
+            return status_code.CRITICAL, msg
+
+
+def convert_to_port(string):
+    """Convert string to valid port."""
+    MIN_PORT_NUMBER = 0
+    MAX_PORT_NUMBER = 65535
+    error_msg = 'Unsigned short integer required, {string} given.'.format(string=string)
+    try:
+        value = int(string)
+    except ValueError:
+        raise argparse.ArgumentTypeError(error_msg)
+    if value < MIN_PORT_NUMBER or value > MAX_PORT_NUMBER:
+        raise argparse.ArgumentTypeError(error_msg)
+    return value
+
+
+def check_run_on_first_broker(broker_list, broker_id, data_path):
+    """Returns true if the first broker in broker_list the same as in args."""
+    broker_id = broker_id if broker_id != -1 else get_broker_id(data_path)
+
+    brokers = sorted(broker_list.keys())
+
+    return broker_id == brokers[0]
+
+
+def get_under_replicated_from_broker(host, port):
+    """Query one broker for info about under replicated partitions.
+
+    :param string host: hostname of broker
+    :param int port: jolokia port for query under replicated partitions number
+    :returns int: number of under replicated partitions for the broker
+    """
+    UNDER_REPL_KEY = 'kafka.cluster:type=Partition,name=UnderReplicated,topic=*,partition=*'
+    JOLOKIA_PREFIX = 'jolokia'
+
+    url = 'http://{host}:{port}/{prefix}/read/{key}'.format(
+        host=host,
+        port=port,
+        prefix=JOLOKIA_PREFIX,
+        key=UNDER_REPL_KEY,
+    )
+
+    response = requests.get(url)
+
+    response.raise_for_status()
+
+    under_replicated = json.loads(response.content)['value']
+    return sum([int(val['Value']) for val in under_replicated.values()])
```

Comment:
I think that the information about what partitions are under replicated is quite important. I would rather to keep a map and in the result give the information about under-replicated partitions for each broker id.